src/core/electrostatics_magnetostatics/elc.cpp

```diff
@@ -247,6 +247,16 @@ static void add_dipole_force(const ParticleRange &particles) {
   gblcblk[2] = 0; // sum q_i
 
   for (auto const &p : local_particles) {
+    /* check if charged particles are in valid region */
+    if ((p.p.q != 0) && ((p.r.p[2] > elc_params.h) || (p.r.p[2] < 0))) {
```

Comment:
why are you checking for the same condition in the next line?

Reply:
Well, only part of the same condition. The outer condition checks whether there is an error, at all. The inner condition to distinguish between the two different directions. I could do two independent checks. But then I would have to check ```q!=0``` twice. Or we could leave away the information about how far the particle is in the forbidden region, then I could leave the inner condition away.

Reply:
what about checking for the charge in the outer conditional, and using `if(p.r.p[2] > elc_params.h) ... else if(p.r.p[2] < 0))` in the inner conditional?

Reply:
i think this should have a name and should be moved out to a function. the current implementation is happening in the function ```add_dipole_force```.